Source/Tests/LinuxSyscalls/x32/Syscalls.h

```diff
@@ -100,20 +102,42 @@ bool RegisterSyscall(int num, int32_t HostSyscallNumber, const char *name, F f){
   struct impl_##name { \
     impl_##name() \
     { \
-      FEX::HLE::x32::RegisterSyscall(x32::SYSCALL_x86_##name, ~0, #name, lambda); \
+      FEX::HLE::x32::RegisterSyscall(x32::SYSCALL_x86_##name, ~0, FEXCore::IR::SyscallFlags::DEFAULT, #name, lambda); \
     } } impl_##name
 
 // Registers syscall for 32bit only
 #define REGISTER_SYSCALL_IMPL_X32_PASS(name, lambda) \
   struct impl_##name { \
     impl_##name() \
     { \
-      FEX::HLE::x32::RegisterSyscall(x32::SYSCALL_x86_##name, SYSCALL_DEF(name), #name, lambda); \
+      FEX::HLE::x32::RegisterSyscall(x32::SYSCALL_x86_##name, SYSCALL_DEF(name), FEXCore::IR::SyscallFlags::DEFAULT, #name, lambda); \
     } } impl_##name
 
 #define REGISTER_SYSCALL_IMPL_X32_PASS_MANUAL(name, hostname, lambda) \
   struct impl_##name { \
     impl_##name() \
     { \
-      FEX::HLE::x32::RegisterSyscall(x32::SYSCALL_x86_##name, SYSCALL_DEF(hostname), #name, lambda); \
+      FEX::HLE::x32::RegisterSyscall(x32::SYSCALL_x86_##name, SYSCALL_DEF(hostname), FEXCore::IR::SyscallFlags::DEFAULT, #name, lambda); \
+    } } impl_##name
+
+
+#define REGISTER_SYSCALL_IMPL_X32_FLAGS(name, flags, lambda) \
+  struct impl_##name { \
+    impl_##name() \
+    { \
+      FEX::HLE::x32::RegisterSyscall(x32::SYSCALL_x86_##name, ~0, flags, #name, lambda); \
+    } } impl_##name
+
+#define REGISTER_SYSCALL_IMPL_X32_PASS_FLAGS(name, flags, lambda) \
```

Comment:
I'm not a fan of the increasing number of these macros. What do they do in the first place compared to just calling `FEX::HLE::x32::RegisterSyscall` directly?

Can we implement this functionality using more unified syntax so we don't need 6 macros to do essentially the same thing?

Reply:
Maybe something like this could work? The stringification is still done in a macro, but everything else in regular functions. Call sites would become something like `SYSCALL_IMPL(mmap).RegisterSyscall(FEXCore::IR::SyscallFlags::OPTIMIZETHROUGH | FEXCore::IR::SyscallFlags::NOSYNCSTATEONENTRY, LAMBDA)`

```cpp
#define SYSCALL_IMPL(name) \
    struct {
        template<class F>
        bool RegisterSyscall(int num, int32_t HostSyscallNumber, F f){
            return FEX::HLE::x64::RegisterSyscall(x64::SYSCALL_x64_##name, SYSCALL_DEF(name), FEXCore::IR::SyscallFlags::DEFAULT, #name, std::move(f));
        }

        template<class F>
        bool RegisterSyscall(int num, int32_t HostSyscallNumber, SysCallFlags flags, F f){
            return FEX::HLE::x64::RegisterSyscall(x64::SYSCALL_x64_##name, SYSCALL_DEF(name), flags, #name, std::move(f));
        }
    } impl_##name
```

We can use the same name for member functions that differ only in strongly typed parameters - for others I'd add a suffix like you're doing with the macros already.

Reply:
How would that work in the face of `Templates cannot be declared inside of a local class [template_inside_local_class]`?

Reply:
C++ was a mistake 😩

I'm sure there's a good way to do this, but let's worry about it another day.